src/test/java/com/puppycrawl/tools/checkstyle/api/FileTextTest.java

```diff
@@ -199,4 +199,18 @@ public void testCharsetAfterCopyConstructor() throws IOException {
         assertWithMessage("Should not be null")
                 .that(copy.getCharset()).isNotNull();
     }
+
+    @Test
+    public void testMalformedInputIgnored() throws IOException {
+        final Charset charset = StandardCharsets.US_ASCII;
+        final String filepath = getPath("InputFileTextMalformedInput.txt");
+        final FileText fileText = new FileText(new File(filepath), charset.name());
+        final CharSequence fullText = fileText.getFullText();
+        assertWithMessage("Should not be null")
+                .that(fullText)
+                .isNotNull();
+        assertWithMessage("Should not contain malformed text")
+                .that(fullText)
+                .isNotEqualTo("some.class.passed");
```

Comment:
If just part of file is dropped, we can have Input based test to make sure thet message contains whole name.

After extra thought.... Why we allow our file loader to drop some content?? Should we just remove code as pitest suggests?
Please review Git history of this line. When and why it was added ?
If is too far in history and without reason, it might be a time before we started to support utf8 source files.

Reply:
@romani as far I can go back in history this was implemented in around 2010.
pr:- https://github.com/checkstyle/checkstyle/commit/3cd6646941eca3da7d3b9a221c88e4c1d214876e

> Should we just remove code as pitest suggests?

yes, there would be no issue

Reply:
Please just remove this code, I think we do not need it now.

Reply:
done

Reply:
Semaphore is failing but i am not able to see why it is failing

Reply:
I added you to project in semaphore, please recheck access now

Reply:
We need to update our sevntu project to skip images, it will be good example what users should do in their projects.

Reply:
@romani I think removal of this 2 might create issue. because we can update the project to skip image it is fine.

but after go through all the error, one Java file is also creating exception. 
https://checkstyle.semaphoreci.com/jobs/18968975-0295-43b1-9f3e-5e5fc8c801d2

The file is https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/sevntu-checks/src/test/resources/com/github/sevntu/checkstyle/checks/design/InputHideUtilityClassConstructorCheck4.java

and the issue is causing 
https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/720cdaad71971a2c88f6ccb888ea0a6efa65e40b/sevntu-checks/src/test/resources/com/github/sevntu/checkstyle/checks/design/InputHideUtilityClassConstructorCheck4.java#L9
over here in this file.

I am not sure that this both are useful or not but removal might create issue 
```
            decoder.onMalformedInput(CodingErrorAction.REPLACE);
            decoder.onUnmappableCharacter(CodingErrorAction.REPLACE);
```

please give your suggestion

Reply:
@romani ping

Reply:
@romani ping

Reply:
@romani ping

Reply:
It is problem in our code but it is separate problem.
Please add input with such javadoc, so survival will be killed.
And we will create issue to investigate why we need this line, it will be issue for outside of summer project.

It might be a problem only for javadoc content.

Reply:
> decoder.onUnmappableCharacter(CodingErrorAction.REPLACE);

@Kevin222004 , after  thinking a bit more on this .....
https://docs.oracle.com/javase/8/docs/technotes/guides/intl/encoding.doc.html
If user set in config charset - `ASCII` but in file we meet character from Turkish `Cp857`, there is definetely is hard situation for parser, and probably dropping characters is only way out, as it is mess and we do not know how to interpretate this character.

usually people goes to lanuage specific chacters in String literals or comments.
We just need a test that shows that we do drop unknown characters and user will not be able to detect them.
we can use https://checkstyle.org/checks/coding/illegaltokentext.html#IllegalTokenText or https://checkstyle.org/checks/javadoc/summaryjavadoc.html#SummaryJavadoc (to check illegal pattern)

Reply:
> decoder.onMalformedInput(CodingErrorAction.REPLACE);

we can simply supply to validation two files: txt and png. With config on file size or even file name validation. PNG file should be ignored due to this decoder settings.
We need separate issue (outside of summer) to remove it or consider it to be customizable to let users control explicitly what they are validating.

Reply:
> decoder.onUnmappableCharacter(CodingErrorAction.REPLACE);
 
This code looks useless to me because Maybe all unmappable characters are replaced by `decoder.onMalformedInput(CodingErrorAction.REPLACE);`

I think in test .png is not required as we already add java file.

you can see https://checkstyle.semaphoreci.com/workflows/4cd4e18a-235c-4f0d-8613-ff5b20f6754a?pipeline_id=c1d7aa95-a1a6-4fb0-b63a-6e6f3718b102 last time seventu was giving error for png which is resolved without unmappable character.

Reply:
> decoder.onUnmappableCharacter(CodingErrorAction.REPLACE);
 
This code looks useless to me because Maybe all unmappable characters are replaced by `decoder.onMalformedInput(CodingErrorAction.REPLACE);`

I think in test .png is not required as we already add java file.

you can see https://checkstyle.semaphoreci.com/workflows/4cd4e18a-235c-4f0d-8613-ff5b20f6754a?pipeline_id=c1d7aa95-a1a6-4fb0-b63a-6e6f3718b102 last time seventu was giving error for png which is resolved without unmappable character.

Reply:
@Vyom-Yadav , please help me to cover this request. I am on vacation with not working laptop, this request needs laptop time to experiment.

Reply:
both Mutation has been killed

Reply:
I think no check is getting affected by this  https://github.com/checkstyle/checkstyle/pull/13181#discussion_r1249920873 
as per my testing on local by adding this text https://github.com/checkstyle/checkstyle/pull/13181#discussion_r1261765695

Reply:
@Kevin222004 For unmappableCharacter, you can try LineLength check on the file I created above. @romani I think we can tolerate an input file being created using code in this case.

Reply:
I don't see reason to generate file, we can use simple and explicit file with non ASCII symbols to show how it will work if user decides to play with encoding

Reply:
Alright, @Kevin222004, create a file with the encoding `IBM01140` manually, and it should have text `Ø`.

Then you can use the following test:
```java
@Test                                                                                            
public void testDecoder() throws Exception {                                                     
    final Charset charset = Charset.forName("IBM1098");                                          
                                                                                                 
    final File file = new File(getPath("InputFileTextCharset.txt"));                             
                                                                                                 
    final FileText fileText = new FileText(file, charset.name());                                
    assertWithMessage("Expected and actual text differ")                                          
            .that(fileText.get(0))                                                                
            .isEqualTo("�");                                                                     
}                                                                                                
```